### PR TITLE
cached cv info for distributed workers

### PR DESCRIPTION
**Pull request by akhileshh**

### CODE REVIEW

File: pychunkedgraph/graph/meta.py [hunk 1]
```diff
@@ -80,7 +84,20 @@ def custom_data(self):
     def ws_cv(self):
         if self._ws_cv:
             return self._ws_cv
-        self._ws_cv = CloudVolume(self._data_source.WATERSHED)
+
+        try:
+            # try reading a cached info file for distributed workers
+            # useful to avoid md5 errors on high gcs load
+            cache_key = f"{self.graph_config.ID}:cv_info_cached"
```
Comment (sdorkenw) [inline pychunkedgraph/graph/meta.py:91]: Define this before the `try` statement. I would add "watershed" to the key to make this unambiguous

File: pychunkedgraph/graph/meta.py [hunk 2]
```diff
@@ -80,7 +84,20 @@ def custom_data(self):
...
+            redis = get_redis_connection()
+            cached_info = json.loads(redis.get(cache_key))
+            self._ws_cv = CloudVolume(self._data_source.WATERSHED, info=cached_info)
+        except Exception:
```
Comment (sdorkenw) [inline pychunkedgraph/graph/meta.py:95]: this could be better handled. Isnt there a way to check if the key exists in redis?

**Review by sdorkenw — CHANGES_REQUESTED** [review]

File: pychunkedgraph/graph/meta.py (hunk 1 above)
Comment (akhileshh) [reply pychunkedgraph/graph/meta.py:91]: The cache key is only used within the try block, no need to define it outside. Will add `watershed`.

File: pychunkedgraph/graph/meta.py (hunk 2 above)
Comment (akhileshh) [reply pychunkedgraph/graph/meta.py:95]: This handles both redis existence and json parsing issues, we want it to work in both cases so if either fails just reads from info file.

Comment (akhileshh) [reply pychunkedgraph/graph/meta.py:95]: And if a redis connection is even available.

Comment (akhileshh) [reply pychunkedgraph/graph/meta.py:95]: When running locally, for instance.

Comment (sdorkenw) [reply pychunkedgraph/graph/meta.py:95]: It's used in the Exception clause and that assumes that the failure happened before the definition

**Review by sdorkenw — CHANGES_REQUESTED** [review]

File: pychunkedgraph/graph/meta.py (hunk 2 above)
Comment (akhileshh) [reply pychunkedgraph/graph/meta.py:95]: Yes, but `try` is executed first, so it will already have been defined when it reaches `except`.

**Pull request merged**